x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all/rules_table_filters/tags_filter_popover.tsx

```diff
@@ -41,9 +41,11 @@ export const TagsFilterPopoverComponent = ({
 }: TagsFilterPopoverProps) => {
   const [isTagPopoverOpen, setIsTagPopoverOpen] = useState(false);
   const [selectedTags, setSelectedTags] = useState<string[]>([]);
+  const didMountRef = useRef(false);
 
   useEffect(() => {
-    onSelectedTagsChanged(selectedTags);
+    if (didMountRef.current) onSelectedTagsChanged(selectedTags);
```

Comment:
here we can just compare if our `selectedTags` is Not equal to our prop tags

Reply:
but our prop tags are all the tags and our selected tags are the selected ones. if they are equal, we need to know that to know what is selected